ClassRegistration/ClassRegistration.App/Controllers/EnrollmentController.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+using ClassRegistration.App.Interfaces;
+using ClassRegistration.App.Repositories;
+using ClassRegistration.DataAccess.Entity;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Security.Cryptography.X509Certificates;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace ClassRegistration.App.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EnrollmentController : ControllerBase
+    {
+
+        /// <summary>
+        /// private IEnrollment field.
+        /// </summary>
+        private IEnrollmentRepo _enrollmentRepository;
+
+        
+        /// <summary>
+        /// Constructor initializes IEnrollment field.
+        /// </summary>
+        /// <param name="enrollmentRepo"></param>
+        public EnrollmentController(IEnrollmentRepo enrollmentRepo)
+        {
+            this._enrollmentRepository = enrollmentRepo;
+
+        }
+        
+       
+
+        /// <summary>
+        /// This method returns the total credits of a student with a specified ID and the term
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="term"></param>
+        /// <returns></returns>
+
+        // GET api/<EnrollmentController>/id
+        [HttpGet("{id}/{term}")]
+        public Object GetTotalCredits(int id,string term)
+        {
+            //return Ok(_enrollmentRepo.GetTotalCredits(id));
+
+            int FallMinimumCredits = 20;  //fall minimum credits
+            int WinterMinimumCredits = 8;  //winter minimum credits
+            int SummerMinCredits = 8;       //summer minimum credits
+
+           
+            int totalCredits = _enrollmentRepository.GetCredits(id, term);
+
+            if(totalCredits >= FallMinimumCredits && term == "Fall")
+            {
+                return Ok(totalCredits);
+
+            }
+            else if(totalCredits >= WinterMinimumCredits && term == "Winter" )
+            {
+                return Ok(totalCredits);
+
+
+            }
+
+            else if(totalCredits >= SummerMinCredits && term == "Summer")
+            {
+                return Ok(totalCredits);
+
+
+            }
+
+            return NotFound();

```

Comment:
To add some extra validation, we probably want to separate some of these checks. Check if the term is "Summer", "Fall", "Winter", then return NotFound if it isn't one of those options. Next, check that the total credits fits the minimum requirement, and return Ok if it does, otherwise return something that tells the client that there was an error due to minimum credits requirments. I would suggest returning a custom Json object here

Reply:
Resolved